datalad/support/tests/test_parallel.py

```diff
@@ -267,6 +267,31 @@ def worker():
     assert_equal(v1, v2)
 
 
+@with_tempfile(mkdir=True)
+def test_parallel_singletons(topd=None):
```

Comment:
```suggestion
def test_parallel_flyweights(topd=None):
```

Not trying to be too nitpicking here, but I find the incorrect use of pattern names a bit confusing.

Reply:
;-) ok, I will change while merging locally / pushing, but my reply could be a neat-picking that such name describes better what this test is intended to ensure -- use of singleton factory method `Flyweight.__call__` which could not guarantee its promise of providing a singleton without such locking.